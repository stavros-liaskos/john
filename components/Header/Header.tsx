import React from 'react';
import DarkMode from '../DarkMode/DarkMode';
import { useUser } from '@auth0/nextjs-auth0';

const Header: React.FunctionComponent = () => {
  const { user } = useUser();

  return (
    <header className="flex justify-between basis-14 rr-border border-b-2">
      <DarkMode />

      {user && (
        <a href="/api/auth/logout">
          <button className="btn btn-small lg:ml-8 mx-3 my-2">Logout</button>
        </a>
      )}
    </header>
  );
};
Header.whyDidYouRender = true;
export default Header;
