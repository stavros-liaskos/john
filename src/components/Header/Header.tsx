import React from 'react';
import DarkMode from '../DarkMode/DarkMode';
import Link from 'next/link';

const Header: React.FunctionComponent = () => (
  <header className="rr-column justify-center flex-none h-14 rr-border border-b-2">
    <div className="flex justify-between items-center w-full lg:w-9/12">
      <DarkMode />

      <Link href="/api/auth/logout">
        <button className="btn btn-small">Logout</button>
      </Link>
    </div>
  </header>
);
Header.whyDidYouRender = false;
export default Header;
