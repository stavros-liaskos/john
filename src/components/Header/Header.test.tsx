import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';
import { UserProvider } from '@auth0/nextjs-auth0/client';

describe('Header', () => {
  it('hides Logout btn when user is logged out', () => {
    const { findByText } = render(
      <UserProvider>
        <Header />
      </UserProvider>,
    );

    expect(findByText('Logout')).toMatchObject({});
  });

  it('shows Logout btn when user is logged in', async () => {
    const component = render(
      <UserProvider user={{ user: 'john.doe' }}>
        <Header />
      </UserProvider>,
    );

    expect(component.container).toMatchSnapshot();
  });
});
