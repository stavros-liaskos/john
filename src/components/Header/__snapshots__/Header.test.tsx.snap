// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header shows Logout btn when user is logged in 1`] = `
<div>
  <header
    class="rr-column justify-center flex-none h-14 rr-border border-b-2"
  >
    <div
      class="flex justify-between items-center w-full lg:w-9/12"
    >
      <button
        aria-label="toggleDarkMode"
        data-testid="moon"
      >
        <svg
          class="h-8 w-8 text-gh-dark"
          fill="none"
          role="img"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 12.79A9 9 0 1111.21 3 7 7 0 0021 12.79z"
          />
        </svg>
      </button>
      <a
        href="/api/auth/logout"
      >
        <button
          class="btn btn-small"
        >
          Logout
        </button>
      </a>
    </div>
  </header>
</div>
`;
